aws/resource_aws_ec2_client_vpn_endpoint.go

```diff
@@ -0,0 +1,347 @@
+	d.Set("client_cidr_block", result.ClientVpnEndpoints[0].ClientCidrBlock)
+	d.Set("server_certificate_arn", result.ClientVpnEndpoints[0].ServerCertificateArn)
+	d.Set("transport_protocol", result.ClientVpnEndpoints[0].TransportProtocol)
+	d.Set("dns_name", result.ClientVpnEndpoints[0].DnsName)
+	d.Set("status", result.ClientVpnEndpoints[0].Status)
+
+	err = d.Set("authentication_options", flattenAuthOptsConfig(result.ClientVpnEndpoints[0].AuthenticationOptions))
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("connection_log_options", flattenConnLoggingConfig(result.ClientVpnEndpoints[0].ConnectionLogOptions))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsEc2ClientVpnEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	_, err := conn.DeleteClientVpnEndpoint(&ec2.DeleteClientVpnEndpointInput{
+		ClientVpnEndpointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return fmt.Errorf("Error deleting Client VPN endpoint: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsEc2ClientVpnEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.ModifyClientVpnEndpointInput{
+		ClientVpnEndpointId: aws.String(d.Id()),
+	}
+
+	if d.HasChange("description") {
+		req.Description = aws.String(d.Get("description").(string))
+	}
+
+	if d.HasChange("dns_servers") {
+		dnsValue := expandStringList(d.Get("dns_servers").(*schema.Set).List())
+		var enabledValue *bool
+
+		if len(dnsValue) > 0 {
+			enabledValue = aws.Bool(true)
+		} else {
+			enabledValue = aws.Bool(false)
+		}
+
+		dnsMod := &ec2.DnsServersOptionsModifyStructure{
+			CustomDnsServers: dnsValue,
+			Enabled:          enabledValue,
+		}
+		req.DnsServers = dnsMod
+	}
+
+	if d.HasChange("server_certificate_arn") {
+		req.ServerCertificateArn = aws.String(d.Get("server_certificate_arn").(string))
+	}
+
+	if d.HasChange("connection_log_options") {
+		if v, ok := d.GetOk("connection_log_options"); ok {
+			var connLogRequest *ec2.ConnectionLogOptions
+			connSet := v.(*schema.Set)
+
+			for _, connObject := range connSet.List() {
+				connData := connObject.(map[string]interface{})
+
+				connLogRequest = expandEc2ConnectionLogOptionsRequest(connData)
+			}
+
+			req.ConnectionLogOptions = connLogRequest
+		}
+	}
+
+	_, err := conn.ModifyClientVpnEndpoint(req)
+	if err != nil {
+		return fmt.Errorf("Error modifying Client VPN endpoint: %s", err)
+	}
+
+	return resourceAwsEc2ClientVpnEndpointRead(d, meta)
+}
+
+func expandEc2ConnectionLogOptionsRequest(data map[string]interface{}) *ec2.ConnectionLogOptions {
+	req := &ec2.ConnectionLogOptions{
+		Enabled: aws.Bool(data["enabled"].(bool)),
+	}
+
+	if data["enabled"].(bool) == true && data["cloudwatch_log_group"].(string) != "" {
+		req.CloudwatchLogGroup = aws.String(data["cloudwatch_log_group"].(string))
+	}
+
+	if data["enabled"].(bool) == true && data["cloudwatch_log_stream"].(string) != "" {
```

Comment:
Nit: `gosimple` linter (being enabled soon) will flag this to remove the `== true` since its redundant

```suggestion
	if data["enabled"].(bool) && data["cloudwatch_log_stream"].(string) != "" {
```